# Global launch parameters (not node-specific)
/**:
  ros__parameters:
    use_sim_time: false

# --- Model Nodes ---
# Franka Model Node Parameters
franka_model_node:
  ros__parameters:
    rim_axis: "z"
    update_freq: 100.0
    input_topic: "/fr3/franka_robot_state_broadcaster/robot_state"
    output_topic: "/fr3_model"
    robot_urdf_filename: "fr3_franka_hand.urdf"
    vel_thres: 0.005
    log:
      enabled: false
      period: 5.0
      timing: true

# Franka RIM Node Parameters  
franka_rim_node:
  ros__parameters:
    rim_axis: "z"
    update_freq: 10.0
    interface_stiffness: 500.0
    interface_damping: 40.0
    log:
      enabled: false
      period: 5.0
      timing: true

# --- DelayRIM ---
delay_rim_node:
  ros__parameters:
    rim_axis: "z"
    update_freq: 500.0 # 1000.0
    delay_compensation_method: "zoh"
    force_scaling: 0.1
    max_workers: 8
    i3_position_scale: 1.5
    # topics
    rim_topic: "/fr3_rim_delayed"
    target_twist_topic_name: "/fr3/target_twist"
    target_pose_topic_name: "/fr3/target_pose"
    interface_force_topic_name: "/fr3/interface_force"
    i3_pose_topic_name: "/i3/pose"
    i3_twist_topic_name: "/i3/twist"
    i3_wrench_topic_name: "/i3/wrench"
    # logs
    log:
      enabled: true
      period: 1.0
      timing: true
      delay_rim: true


# --- Inverse3 Node ---
# Real
inverse3_node:
  ros__parameters:
    frequency: 1000.0
    scale: 1.0  # scale between i3 position and published position 
    disable_forces: true

# Sim
i3_sim_node:
  ros__parameters:
    publish_frequency: 1000.0
    sine_frequency: 0.2
    amplitude: 0.1
    n_cycles: 3
    use_start_srv: false
    start_time: 2.0

# --- Viz ---
# Visualization Node Parameters
rim_vis_node:
  ros__parameters:
    i3_sphere_size: 0.03
    rim_sphere_size: 0.03

# --- Network Delays ---
rim_msg_delay:
  ros__parameters:
    input_topic: "/fr3_rim"
    output_topic: "/fr3_rim_delayed" 
    delay: 0
    message_type: "arm_interfaces/msg/FrankaRIM"

ee_cmd_delay:
  ros__parameters:
    input_topic: "/osc_pd_controller/goal_pre_delay"
    output_topic: "/osc_pd_controller/goal"
    delay: 0
    message_type: "geometry_msgs/msg/PointStamped"
